### PR TITLE
feat(solar): Shadow & Sun overlay + landing widget (reference-based scaling)

### PR DESCRIPTION
**Pull request by shahiakhilesh1304**
This pull request introduces a new "Shadow & Sun Angle" calculator feature to the landing page and contribution graph, allowing users to estimate sun position and shadow length based on their location and object height. It includes reusable logic for solar calculations, UI components for user interaction, and associated styling. Additionally, it adds automated tests for the core calculation logic.

**New Shadow & Sun Angle Feature**

* Added a `shadow-calculator` widget to the landing page, allowing users to enter an object height, detect their location, and view real-time sun altitude, azimuth, and shadow length. Includes full lifecycle management and accessibility considerations. [[1]](diffhunk://#diff-746e673e0def8b3b4d6cd52e741eccf0d976948b0d33f9027bd325bddcdb468eR48) [[2]](diffhunk://#diff-746e673e0def8b3b4d6cd52e741eccf0d976948b0d33f9027bd325bddcdb468eR113) [[3]](diffhunk://#diff-746e673e0def8b3b4d6cd52e741eccf0d976948b0d33f9027bd325bddcdb468eR316-R317) [[4]](diffhunk://#diff-746e673e0def8b3b4d6cd52e741eccf0d976948b0d33f9027bd325bddcdb468eR346) [[5]](diffhunk://#diff-746e673e0def8b3b4d6cd52e741eccf0d976948b0d33f9027bd325bddcdb468eR568-R572) [[6]](diffhunk://#diff-b17aa7a9aeaca0e73916b12260e740081f612482453b48d1e3401ca17dfff296R1-R191)
* Styled the new landing page shadow calculator section and its controls to match the application's material-inspired design, with responsive layout for mobile.

**Core Calculation Logic**

* Implemented `calculateShadowInfo` in `src/core/solar/shadow.ts` to compute sun position and shadow length for a given object height, location, and time. Handles edge cases (e.g., sun below horizon).
* Added unit tests for `calculateShadowInfo` covering input validation, sun below horizon, and typical use cases.

**Contribution Graph Integration**

* Introduced a `shadow-overlay` visual widget for the contribution graph time page, displaying real-time shadow simulation and metrics based on user input and geolocation.
* Integrated the shadow overlay into the contribution graph renderer lifecycle, ensuring proper mount and cleanup. [[1]](diffhunk://#diff-47839571637318aff1dece52c00d9fe0feedc13df68fef64293c5414c57cea7bR39-R50) [[2]](diffhunk://#diff-47839571637318aff1dece52c00d9fe0feedc13df68fef64293c5414c57cea7bR105-R106)

### CODE REVIEW

File: src/components/landing-page/shadow-calculator.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      return;
+    }
+
+    const now = new Date();
+    const result = calculateShadowInfo({
+      heightMeters,
+      latitude: coords.latitude,
+      longitude: coords.longitude,
+      date: now,
+    });
```
[the remark this answers is not in the document]
Comment (shahiakhilesh1304) [reply src/components/landing-page/shadow-calculator.ts]: Try catch is added

File: src/themes/contribution-graph/renderers/shadow-overlay.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  destroy(): void;
+}
+
+const MIN_HEIGHT = 0.05;
+const MAX_HEIGHT = 5;
+const REFERENCE_HEIGHT_METERS = 1;
+const MIN_SCALE = 0.25; // 0.25x of reference visually
+const MAX_SCALE = 3;    // 3x of reference visually
+
+export function createShadowOverlay(parent: HTMLElement): ShadowOverlayController {
```
[the remark this answers is not in the document]
Comment (shahiakhilesh1304) [reply src/themes/contribution-graph/renderers/shadow-overlay.ts]: Shadow length scale factor and max shadow pixel is added
